prow/config/tide.go

```diff
@@ -64,6 +64,11 @@ type Tide struct {
 	// controller to handle org/repo:branch pools. Defaults to 20. Needs to be a
 	// positive number.
 	MaxGoroutines int `json:"max_goroutines,omitempty"`
+
+	// BranchProtection specifies branch protection options.
+	// Initially it is used for allowing to merge a PR when required checks only are passing,
+	// but it can be extended with more of the github features.
+	BranchProtection TideProtection `json:"branchProtection,omitempty"`
```

Comment:
I don't understand why this is necessary. Tide can just read the branch-protection options?

Reply:
don't you think one would still want to prevent merging when one of the non required checks is failing ?If not then, you are right.

Reply:
@BenTheElder @cjwagner what do you think ? I am fine either way.

Reply:
This makes more sense as an enum/boolean field on each [`TideQuery`](https://github.com/kubernetes/test-infra/blob/9fc2342b69a911052845f42a81f6409ef4553d4a/prow/config/tide.go#L90)

Just like I can toss a `reviewApprovedRequired: true` into a query I'd expect to be able to toss a `ignoreOptionalContexts: true` [here](https://github.com/kubernetes/test-infra/blob/master/prow/config.yaml#L163)

Reply:
I was gonna do that first, but then since you can't really query it I decided to use a separate option, but I really like your idea.

Reply:
actually looking at the implementation, you can define multiple queries per PR, and therefore you could define a query with ignoreOptionalContexts enabled and another disabled which could be a nice option, however when we go in the subpool we loose the query information. We could hack the code to make this happen, but that's ugly. The more I think about it, the more I think we should have an option in the branch protection config. What do you think ?